### ISSUE TITLE
DL3041 warnings when package name is not hyphen-separated from version

### ISSUE THREAD

**Issue by davewhiteley** [hadolint/hadolint#981]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Preferably there would be no **DL3041** warnings when a package is `dnf install` 'd that is **not** hyphen-separated from its pinned version, especially when such packages are unavailable.

Example: `python-3.11` does not cause a **DL3041** warning, but in not available; `python3.11` is available, but causes warnings.

### Actual behavior
```bash
DL3041 warning: Specify version with `dnf install -y <package>-<version>`.
```

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
$ hadolint --version
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile**

```Dockerfile
ARG python_version=3.11
RUN microdnf install -y python${python_version} \
                        python${python_version}-pip \
```

**Additional environment details (OS, stack version, etc.)**

Base image is originally based on [IronBank RedHat UBI 9.2 minimal](https://repo1.dso.mil/dsop/redhat/ubi/9.x/ubi9-minimal)

```bash
$ cat /etc/os-release | grep PRETTY_NAME
PRETTY_NAME="Red Hat Enterprise Linux 9.2 (Plow)"
```

```bash
$ microdnf repolist

repo id                               repo name
ubi-9-appstream-rpms                  Red Hat Universal Base Image 9 (RPMs) - AppStream
ubi-9-baseos-rpms                     Red Hat Universal Base Image 9 (RPMs) - BaseOS
ubi-9-codeready-builder               Red Hat Universal Base Image 9 (RPMs) - CodeReady Builder
```

**Comment by maximiliankolb** [comment]
Hi @davewhiteley 

I believe this is a misunderstanding: The name of the package is "python3.11", the version and release is `3.11.9-2.el8_10`.

Reproducer:

````
$ podman run --rm -it almalinux:8.10

# dnf makecache
...
# dnf install python3.11
...
 Package                                                       Architecture                             Version                                             Repository                                   Size
==============================================================================================================================================================================================================
Installing:
 python3.11                                                    x86_64                                   3.11.9-2.el8_10                                     appstream                                    29 k
Installing dependencies:
...
````

This means that it's not a bug in hadolint. If you install `python3.11`, this is indeed not versioned/pinned, and therefore hadolint will complain.

You can verify this by using `dnf info`:

````
# dnf info python3.11
...
Name         : python3.11
Version      : 3.11.9
Release      : 2.el8_10
Architecture : x86_64
Size         : 29 k
Source       : python3.11-3.11.9-2.el8_10.src.rpm
Repository   : appstream
Summary      : Version 3.11 of the Python interpreter
...
````

version in SPEC file: https://src.fedoraproject.org/rpms/python3.11/blob/rawhide/f/python3.11.spec#_10
